docs/design/2020-06-24-placement-rules-in-sql.md

```diff
@@ -0,0 +1,391 @@
+
+`LABEL`, `ROLE` and `COUNT` configurations are the same as above.
+
+Key format of the partitioned table is in such a format: `t_{partition_id}_r_{pk_value}`. As `partition_id` is part of the prefix of the keys, all data in one partition forms only one key range.
+
+`partition_id` can be inferred from `partition_name`, so the key range can be `t_{partition_id}_` to `t_{partition_id+1}_`.
+
+Placement rules can also be defined on a partitioned table. Because there’s more than one key range for the table, multiple rules should be generated and sent to PD. When placement rules are defined both on the table and its partitions, the rule priorities described later should be applied.
+
+### Unpartitioned index placement
+
+Defining placement of indexes is more complicated, because indexes can be unpartitioned or partitioned. Each case should be considered separately.
+
+The index here can be primary index or secondary index. When `_tidb_rowid` is used for the clustered index instead of the primary key, the primary index is actually an unclustered index. In this case, an index placement statement is applied.
+
+Expression indexes and invisible indexes are also supported, as the key format is the same as normal.
+
+Defining placement of an unpartitioned index:
+
+```sql
+ALTER TABLE table_name ALTER INDEX index_name
+	{ADD | ALTER} PLACEMENT LABEL=label ROLE=role COUNT=count,
+	...
+```
+
+This key format of unpartitioned index is as `t_{table_id}_i_{index_id}_r_{pk_value}`. The key range can be inferred by `table_id` and `index_id`.
+
+### Partitioned index placement
+
+Defining placement of the index in one specific partition:
+
+```sql
+ALTER TABLE table_name ALTER PARTITION partition_name ALTER INDEX index_name
+	{ADD | ALTER} PLACEMENT LABEL=label ROLE=role COUNT=count,
+...
+```
+
+The key format of partitioned index is as `t_{partition_id}_i_{index_id}_r_{pk_value}`. The key range can be inferred by `partition_id` and `index_id`.
+
+When an index is partitioned, defining placement of the whole index at once is not supported. It will involve multiple key ranges, and the scenario of its application is rare.
+
+For example, `t` is a partitioned table and index `idx` is the index on `t`. It’s not supported to do this:
+
+```sql
+ALTER TABLE `t` ALTER INDEX `i` ADD PLACEMENT ...
+```
+
+To alter the placement rule of `idx`, a partition must be specified in the statement.
+
+Currently, global secondary index on partitioned tables is not supported, so it can be ignored for now.
+
+### Sequence placement
+
+Sequence is typically used to allocate ID in INSERT statements, so the placement of sequences affects the latency of INSERT statements.
+
+However, sequence is typically used with cache enabled, which means very few requests are sent to sequence. So defining placement of sequences is not supported for now.
+
+## Drop placement rules
+
+Dropping a placement rule can be achieved by a DROP clause:
+
+```sql
+ALTER TABLE table_name
+	DROP PLACEMENT
```

Comment:
Does this statement drop all the placement rules defined in this table? How to drop a specified placement rule?

Reply:
Good catch. I've updated the doc. Now only dropping a specific rule is supported.